native/game_backend/src/player.rs

```diff
@@ -120,14 +125,30 @@ impl Player {
     }
 
     pub fn add_cooldown(&mut self, skill_key: &String, cooldown_ms: u64) {
-        self.cooldowns.insert(skill_key.to_string(), cooldown_ms);
+        self.cooldowns
+            .entry(skill_key.to_string())
+            .or_insert(cooldown_ms);
+        if self.cooldowns.contains_key(BASIC_SKILL_KEY) {
```

Comment:
I think this won't stop us of depleting a charge when using other skill than the basic either, since it the case where we want to use a skill when the basic_skill is on cooldown is possible

Maybe something like this would do:

```suggestion
        if skill_key == BASIC_SKILL_KEY {
```